src/vivarium/framework/state_machine.py

```diff
@@ -516,7 +521,7 @@ def __init__(
             self.add_states(states)
 
         states_with_initialization_weights = [
-            s for s in self.states if s.initialization_weights != default_initializer
+            s for s in self.states if s.initialization_weights
```

Comment:
Give a more descriptive variable name. Or at the _least_ change it away from "s" b/c that collides w/ pdb's "s" command.